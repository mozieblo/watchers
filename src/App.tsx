import React, { FC } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { ROUTES } from './router/router';
import Home from './pages/home/Home';
import About from './pages/about/About';
import ContactPage from './pages/contactPage/ContactPage';
import { globalTheme } from './styles/theme';

const App: FC = () => {
    const { HOME, ABOUT, CONTACT } = ROUTES;

    return (
        <ThemeProvider theme={globalTheme}>
            <Switch>
                <Route path={HOME} exact component={Home} />
                <Route path={ABOUT} exact component={About} />
                <Route path={CONTACT} exact component={ContactPage} />
                <Redirect to={HOME} />
            </Switch>
        </ThemeProvider>
    );
};

export default App;
