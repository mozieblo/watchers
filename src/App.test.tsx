import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { intlWrapper } from './config/test/intlWrapper';
import App from './App';

test('App: have routing', () => {
    render(
        <BrowserRouter>
            <App />
        </BrowserRouter>,
        { wrapper: intlWrapper }
    );

    expect(<App />).toMatchSnapshot();
    expect(screen.getByText('My React starter.')).toBeInTheDocument();
});
