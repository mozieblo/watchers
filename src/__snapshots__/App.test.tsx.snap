// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: have routing 1`] = `<App />`;
